eth/data.go

```diff
@@ -278,3 +281,23 @@ func validateHex(value string, size int, typ string) (string, error) {
 
 	return value, nil
 }
+
+// RLP returns the Data32 as an RLP-encoded string.
+func (d *Data32) RLP() rlp.Value {
+	return rlp.Value{
+		String: d.String(),
+	}
+}
+
+// Hash returns the keccak256 hash of the Data.
+func (d Data) Hash() Hash {
+	b := d.Bytes()
+	// And feed the bytes into our hash
+	hash := sha3.NewLegacyKeccak256()
+	hash.Write(b)
+	sum := hash.Sum(nil)
+
+	// and finally return the hash as a 0x prefixed string
+	digest := hex.EncodeToString(sum)
+	return Hash("0x" + digest)
```

Comment:
Any tests for these for future-proofing, since they're used as helpers for other functions?

Reply:
Good call, added in 3c234c5.  Also made the hash implementation take an interface so it could be shared across `Data8`, `Data20` etc not just `Data` and added tests for each.